model/labels/labels_stringlabels.go

```diff
@@ -320,6 +320,11 @@ func FromStrings(ss ...string) Labels {
 	return New(ls...)
 }
 
+// FromByteString creates new labels from byte string.
+func FromByteString(b string) Labels {
```

Comment:
The problem here is that you are only adding this for one implementation of `labels.Labels`. Note that this directory has a number of implementation, gated by build tags: `slicelabels`, `dedublabels`, and "none of these two", which is the current default and what you are changing here. With the other implementations, you might run into more trouble. In general, I think we should avoid reverse-engineering the metric name out of the signature. See my thoughts below.

Reply:
Yeah, reverse-engineering the metric name from the signature does seem quite hard, especially across all these implementations. Makes sense to avoid doing that.